### ISSUE TITLE
Rule proposal: `no-await-in-promise-methods`

### ISSUE THREAD

**Issue by Clement398** [sindresorhus/eslint-plugin-unicorn#2257]
### Description

Disallow using `await` in `Promise` method parameters.

### Fail

```js
Promise.all([await promise, anotherPromise]);

Promise.allSettled([await promise, anotherPromise]);

Promise.any([await promise, anotherPromise]);

Promise.race([await promise, anotherPromise]);
```

### Pass

```js
Promise.all([promise, anotherPromise]);

Promise.allSettled([promise, anotherPromise]);

Promise.any([promise, anotherPromise]);

Promise.race([promise, anotherPromise]);
```

**Comment by sindresorhus** [comment]
Same as https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2256#issuecomment-1890700304

**Comment by Clement398** [comment]
It's edited, thanks for the suggestion.
